mllib/src/main/scala/org/apache/spark/mllib/tree/DecisionTree.scala

```diff
@@ -62,8 +64,9 @@ class DecisionTree private[spark] (private val strategy: Strategy, private val s
    */
   @Since("1.2.0")
   def run(input: RDD[LabeledPoint]): DecisionTreeModel = {
-    val rf = new RandomForest(strategy, numTrees = 1, featureSubsetStrategy = "all",
-      seed = seed)
+    val instr = Instrumentation.create(new DecisionTreeClassifier, input)
+    val rf =
+      new RandomForest(strategy, numTrees = 1, featureSubsetStrategy = "all", seed = seed, instr)
```

Comment:
I don't know what's the best way to handle this case: calling `run` from the old mllib api.

Please advise @thunterdb